src/main/java/practice/StreamPractice.java

```diff
@@ -74,7 +105,32 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * We want to reuse our validation in future, so let's write our own impl of Predicate
      * parametrized with Candidate in CandidateValidator.
      */
+
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator candidateValidator = new CandidateValidator();
+        return candidates.stream()
+                .filter(candidate -> candidateValidator.test(candidate))
+                .map(candidate -> candidate.getName())
+                .sorted()
+                .toList();
+    }
+
+    private Predicate<Person> getWorkablePeoplePredicate(int fromAge, int femaleToAge,
+                                                   int maleToAge) {
```

Comment:
```suggestion
    private Predicate<Person> getWorkablePeoplePredicate(
      int fromAge, 
      int femaleToAge,
      int maleToAge
    ) {
```
I prefer this formatting